onadata/apps/api/tests/viewsets/test_project_viewset.py

```diff
@@ -1149,6 +1149,29 @@ def test_project_partial_updates(self):
         self.assertEqual(response.status_code, 200)
         self.assertEqual(project.metadata, json_metadata)
 
+    def test_cache_updated_on_project_update(self):
+        view = ProjectViewSet.as_view({
+            'get': 'retrieve',
+            'patch': 'partial_update'
+        })
+        self._project_create()
+        request = self.factory.get('/', **self.extra)
+        response = view(request, pk=self.project.pk)
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(False, response.data.get("public"))
+
+        projectid = self.project.pk
+        data = {'public': True}
+        request = self.factory.patch('/', data=data, **self.extra)
+        response = view(request, pk=projectid)
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(True, response.data.get("public"))
+
+        request = self.factory.get('/', **self.extra)
+        response = view(request, pk=self.project.pk)
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(True, response.data.get("public"))
```

Comment:
I'd like to think we'd want to test caching by comparing response.data with `cache.get(f'{PROJ_OWNER_CACHE}{project_id}')` which is found in the retrieve action.

Reply:
I added some assertions/comparisons for cached data and response data